engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/DockerOverseer.java

```diff
@@ -58,17 +60,27 @@ public DockerOverseer(
         }));
         pos = stdout.length();
         this.client = client;
-        init.thenAccept(containerId ->
-                executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS)
-        );
+        init.thenAccept(containerId -> {
+            this.containerId = containerId;
+            executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS);
+        });
         this.init = init;
     }
 
     @Override
     public CompletableFuture<ExecutionOutcome> abort() {
-        return result.isDone() ? result : CompletableFuture.completedFuture(
-                new ExecutionOutcome(Stage.ABORT, "Abort is not yet supported for docker analysis", null)
-        );
+        String logs = getExecutionLogs();
+        return CompletableFuture.supplyAsync(() -> {
+            if (result.isDone()) {
+                return result.join();
+            } else {
+                client.stopContainerCmd(containerId).exec();
```

Comment:
Major: If `init` has not yet completed (e.g. the container is still being downloaded), this will fail with NPE and then, container will wind up and proceed to completion, effectively ignoring the abort command.

Reply:
Thanks for catching this up. To handle this case I believe we should wait for its completion before proceeding with the abort logic. We will use the `thenCompose `method to achieve this.